pkg/netstate/netstate.go

```diff
@@ -36,10 +40,24 @@ type DB interface {
 	Delete([]byte) error
 }
 
+func isAuthValid(xAPIKeyBytes []byte) bool {
+	return auth.ValidateAPIKey(string(xAPIKeyBytes))
+}
+
 // Put formats and hands off a file path to be saved to boltdb
 func (s *Server) Put(ctx context.Context, filepath *proto.FilePath) (*proto.PutResponse, error) {
 	s.logger.Debug("entering netstate put")
 
+	xAPIKeyBytes := []byte(filepath.XApiKey)
+	if response := isAuthValid(xAPIKeyBytes); !response {
+		s.logger.Error("unauthorized request")
+		fmt.Println(grpc.Errorf(codes.Unauthenticated, "Invalid API credential"))
+
+		return &proto.PutResponse{
+			Confirmation: "Unauthorized Request",
+		}, nil
+	}
+
```

Comment:
the most i can do with validateAuth is returning the auth. code; every request has a different object that I'm sending in as a parameter. The least common denominator with all the parameters is sending a []byte. From there I can validate the auth and proceed from there. 

If I return nil, err in the functions, I'll get a runtime error because the response is nil.  From what I can see, I'll have to keep the returns as I have them.  If there is a better way to go about it, I'd love to know more.  That's why I didn't return nil, err. 

thanks for your comments.

Reply:
* note that this comment isn't outdated! @jtolds I think we tried doing what you did returning `nil, err` but got the runtime error.